### PR TITLE
Initial Release

### PR DESCRIPTION
**Pull request by bryanlalexander**
This is the first iteration for IAM groups, roles, and policies. Please take a careful look and feel free to request additional roles and groups as they make sense. Currently there are only four groups which include:

- **full-admin** which has full administrator access and requires MFA
- **ops-admin** which has region-restricted administrator access, explicit IAM permissions, and requires MFA
- **resource-admin** which has region-restricted administrator access (excluding IAM, Organizations, sts:AssumeRole), and requires MFA
- **deployer-admin** which has region-restricted administrator access (excluding IAM, Organizations, sts:AssumeRole

It is likely we probably need a group for read-only or similar but I'm open to suggestions.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+variable "password_policy_allow_password_changes" {
+  type        = bool
+  description = "(optional) the boolean value indicating whether to allow users to change their passwords"
+  default     = true
+}
+
+variable "password_policy_max_age_days" {
+  type        = number
+  description = "(optional) the number representing the number of days before a password should expire"
+  default     = 90
```
Comment (brady-gsa) [inline variables.tf:51]: Is 90 based on Microsoft's recommendation of 30 to 90 days? Some would argue this does more harm than good. See https://www.sans.org/security-awareness-training/blog/time-password-expiration-die

File: variables.tf [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+variable "password_policy_password_history_days" {
+  type        = number
+  description = "(optional) the number representing the number of previous passwords to remember"
+  default     = 24
```
Comment (brady-gsa) [inline variables.tf:57]: Similar question here - although I'm not sure which standard recommends 24.

**Review by briankfitzwater — COMMENTED** [review]
The deployer role requires additional IAM privileges. The current circle-deployer user has an inline policy that adds the following:

```
            "Action": [
                "iam:AttachRolePolicy",
                "iam:CreatePolicy",
                "iam:CreatePolicyVersion",
                "iam:CreateRole",
                "iam:DeletePolicy",
                "iam:DeletePolicyVersion",
                "iam:DeleteRole",
                "iam:DetachRolePolicy",
                "iam:GetPolicy",
                "iam:GetPolicyVersion",
                "iam:GetRole",
                "iam:GetRolePolicy",
                "iam:ListAttachedRolePolicies",
                "iam:ListInstanceProfilesForRole",
                "iam:ListPolicyVersions",
                "iam:PassRole",
                "iam:PutRolePolicy",
                "iam:UpdateRoleDescription",
                "iam:AddUserToGroup",
                "iam:AttachGroupPolicy",
                "iam:CreateGroup",
                "iam:DeleteGroup",
                "iam:DeleteGroupPolicy",
                "iam:DetachGroupPolicy",
                "iam:GetGroup",
                "iam:GetGroupPolicy",
                "iam:ListAttachedGroupPolicies",
                "iam:ListGroupPolicies",
                "iam:ListGroups",
                "iam:ListGroupsForUser",
                "iam:PutGroupPolicy",
                "iam:RemoveUserFromGroup",
                "iam:UpdateGroup",
                "iam:UpdateAccountPasswordPolicy",
                "iam:GetAccountPasswordPolicy"
            ],
            "Effect": "Allow",
            "Resource": "*"
        },
```

There is already an AWS maintained ReadOnlyAccess policy (`arn:aws:iam::aws:policy/ReadOnlyAccess`), but a ReadOnly role with that policy attached would be a good idea.

**Comment by bryanlalexander** [comment]
@briankfitzwater This would be for a generic deployer account, not really for our usage mainly just for easy deployment into an empty AWS account.

**Review by briankfitzwater — APPROVED** [review]

**Pull request merged**